### ISSUE TITLE
When writing large messages queue adds latency.

### ISSUE THREAD

**Issue by peter-lawrey** [OpenHFT/Chronicle-Queue#480]
When the start and end of a message are in different chunks, there is latency introduced as it switches back to write the header. 
We could look at ensuring the overlap is used instead of switching.
We could look at cache the last two chunks instead of the last one to avoid this.

BenchmarkMain shows this issue
```
messageSize 65536
transport histogram: 50/90 99/99.9 99.99/99.999 - worst was 0.075 / 0.14  0.38 / 15  555 / 1,380 - 3,990
messageSize 262144
transport histogram: 50/90 99/99.9 99.99 - worst was 0.16 / 0.34  5.6 / 16  645 - 1,350
```

From 64 KB to 256 KB, the 99%ile jumps from 0.38 us to 5.6 us.
It should be possible to avoid this delay.

**Comment by RobAustin** [comment]
This should now be fixed due to the following code which if it is able to 

```
 if (safeCopySize + mappedFile.overlapSize() >= remaining) 
```

will ensure the overlap is used instead of switching.


```
    public MappedBytes write(long writeOffset, RandomDataInput bytes, long readOffset, long length)
            throws BufferOverflowException, BufferUnderflowException {

        long wp = writeOffset;

        if (length > writeRemaining())
            throw new DecoratedBufferOverflowException(
                    String.format("write failed. Length: %d > writeRemaining: %d", length, writeRemaining()));

        long remaining = length;

        acquireNextByteStore(wp, false);

        while (remaining > 0) {

            long safeCopySize = copySize(wp);

            if (safeCopySize + mappedFile.overlapSize() >= remaining) {
                bytesStore.write(wp, bytes, readOffset, remaining);
                return this;
            }

            bytesStore.write(wp, bytes, readOffset, safeCopySize);

            readOffset += safeCopySize;
            wp += safeCopySize;
            remaining -= safeCopySize;

            // move to the next chunk
            acquireNextByteStore0(wp, false);
        }
        return this;
    }
```

**Comment by hft-team-city** [comment]
Released in Chronicle-Queue-5.17.1 (https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.1), BOM-2.17.44 (https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.44)

**Comment by hft-team-city** [comment]
Released in Chronicle-Queue-5.17.1 (https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.1), BOM-2.17.48 (https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.48)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.1](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.1), [BOM-2.17.50](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.50)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.1](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.1), [BOM-2.17.52](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.52)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.1](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.1), [BOM-2.17.53](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.53)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.1](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.1), [BOM-2.17.60](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.60)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.1](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.1), [BOM-2.17.72](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.72)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.1](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.1), [BOM-2.17.74](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.74)
